cargo-near/src/cargo/metadata.rs

```diff
@@ -15,15 +15,17 @@ pub(crate) struct CrateMetadata {
 impl CrateMetadata {
     /// Parses the contract manifest and returns relevant metadata.
     pub fn collect(manifest_path: CargoManifestPath) -> Result<Self> {
-        let (metadata, root_package) = get_cargo_metadata(&manifest_path)?;
+        let (mut metadata, root_package) = get_cargo_metadata(&manifest_path)?;
+        metadata.target_directory = metadata.target_directory.canonicalize_utf8()?;
+        metadata.workspace_root = metadata.workspace_root.canonicalize_utf8()?;
```

Comment:
if we're forcing utf8 here, we might as well change `--out-dir`, `--manifest-path` to support valid `utf8` inputs.

Neither `cargo` nor `rustc` support non-utf8 names/paths.

<img width="359" alt="CleanShot 2022-10-19 at 17 10 42@2x" src="https://user-images.githubusercontent.com/16881812/196700352-6d1a22b0-277f-482d-bcb2-f7c9c8855eba.png">


<img width="422" alt="CleanShot 2022-10-19 at 17 07 30@2x" src="https://user-images.githubusercontent.com/16881812/196699637-ce90215f-c7fe-4bc9-a8b4-8bd8e03a63ba.png">

Reply:
#76